### PR TITLE
Do not provide Intellisense error suggestions in string interpolation

### PR DESCRIPTION
**Pull request by rick-nguyen**
If the parent of the current node is a string interpolation node, we won't provide any error suggestions - suggestions should only be provided inside the island.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/SuggestionHandlers/ErrorNodeSuggestionHandlerBase.cs [hunk 1]
```diff
@@ -23,6 +23,12 @@ internal override bool TryAddSuggestionsForNodeKind(IntellisenseData.Intellisens
                 // ThisItemProperties only in the context of thisItem.
                 var curNode = intellisenseData.CurNode;
 
+                // Do not provide error suggestions in string interpolation scenario
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/SuggestionHandlers/ErrorNodeSuggestionHandlerBase.cs:26]: can you add a bit more detail here? Basically, clarify that if the parent is a stringinterp node and not an island, this node is always a partial string literal, and we shouldn't provide suggestions in string literals.

File: src/tests/Microsoft.PowerFx.Core.Tests/IntellisenseTests/SuggestTest.cs [hunk 2]
```diff
@@ -152,6 +152,7 @@ private string[] SuggestStrings(string expression, PowerFxConfig config, RecordT
         [InlineData("ForAll([0],`|", "ThisRecord", "Value")]
         [InlineData("ForAll(-],|", "ThisRecord")]
         [InlineData("ForAll()~|")]
+        [InlineData("With( {Apples:3}, $\"We {Apples} apples|")]
```
Comment (lesaltzm) [inline src/tests/Microsoft.PowerFx.Core.Tests/IntellisenseTests/SuggestTest.cs:155]: Also add tests of it working correctly within the island, if possible!

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by bryancroteau-MSFT — APPROVED** [review]

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by lesaltzm — APPROVED** [review]
🚢

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Pull request merged**